packages/fx-core/src/core/index.ts

```diff
@@ -885,30 +893,44 @@ class CoreImpl implements Core {
   }
 
   public async writeConfigs(): Promise<Result<null, FxError>> {
-    if (!fs.existsSync(`${this.ctx.root}/.${ConfigFolderName}`)) {
+    const confFolderPath = path.resolve(this.ctx.root, `.${ConfigFolderName}`);
+    if (!fs.existsSync(confFolderPath)) {
       this.ctx.logProvider?.warning(
-        `[Core] writeConfigs() - folder does not exist:${this.ctx.root}/.${ConfigFolderName}`
+        `[Core] writeConfigs() - folder does not exist: ${confFolderPath}`
       );
       return ok(null);
     }
-    try {
-      for (const entry of this.configs.entries()) {
-        const envName = entry[0];
-        const solutionConfig = entry[1];
-        const configJson = mapToJson(solutionConfig);
-        const filePath = `${this.ctx.root}/.${ConfigFolderName}/env.${envName}.json`;
-        const localDataPath = `${this.ctx.root}/.${ConfigFolderName}/${envName}.userdata`;
-        const localData = sperateSecretData(configJson);
-        const content = JSON.stringify(configJson, null, 4);
-        await fs.writeFile(filePath, content);
-        await fs.writeFile(localDataPath, serializeDict(localData));
+    let res: Result<null, FxError> = ok(null);
+    for (let i = 0; i < 5; ++i) {
```

Comment:
Why are we doing this loop 5 times.

Reply:
It's guarantee to make sure config is written successfully